### PR TITLE
Github Actions improvements

### PR DESCRIPTION
**Pull request by NeoRaider**
During debugging of a spurious build failure (that hasn't reoccurred yet...) I found some potential for improvement in our Github Actions setup.

- Set CONFIG_AUTOREMOVE to avoid running out of disk space
  - Do our Jenkins jobs build incrementally or from scratch? If the latter is the case, the flag should be set for Jenkins as well
- Set BUILD_LOG=1 and save logs as separate artifact
  - This step is run even when the build fails
- Some minor script cleanup

### CODE REVIEW

**Comment by mweinelt** [comment]
> Set CONFIG_AUTOREMOVE to avoid running out of disk space
> 
>     * Do our Jenkins jobs build incrementally or from scratch? If the latter is the case, the flag should be set for Jenkins as well

They do incremental builds.

File: contrib/actions/generate-actions.py [hunk 1]
```diff
@@ -25,13 +25,19 @@
...
-        run: bash contrib/actions/install-dependencies.sh
+        run: sudo contrib/actions/install-dependencies.sh
       - name: Build
-        run: bash contrib/actions/run-build.sh {target_name}
+        run: contrib/actions/run-build.sh {target_name}
+      - name: Archive build logs
+        if: ${{{{ !cancelled() }}}}
+        uses: actions/upload-artifact@v1
+        with:
+          name: {target_name}-logs
```
Comment (mweinelt) [inline contrib/actions/generate-actions.py:35]:
```suggestion
          name: {target_name}/logs
```

IMO improves readability

File: contrib/actions/generate-actions.py [hunk 2]
```diff
@@ -25,13 +25,19 @@
...
+          path: openwrt/logs
       - name: Archive build output
         uses: actions/upload-artifact@v1
         with:
-          name: {target_name}
+          name: {target_name}-output
```
Comment (mweinelt) [inline contrib/actions/generate-actions.py:40]:
```suggestion
          name: {target_name}/output
```

File: docs/user/getting_started.rst [hunk 3]
```diff
@@ -208,6 +208,12 @@ GLUON_TARGET
 Special variables
 .................
 
+GLUON_AUTOREMOVE
+  Setting ``GLUON_AUTOREMOVE=1`` enable the ``CONFIG_AUTOREMOVE`` OpenWrt setting, which will delete package build
```
Comment (mweinelt) [inline docs/user/getting_started.rst:212]:
```suggestion
  Setting ``GLUON_AUTOREMOVE=1`` enables the ``CONFIG_AUTOREMOVE`` OpenWrt setting, which will delete package build
```

File: contrib/actions/generate-actions.py (hunk 1 above)
Comment (NeoRaider) [reply contrib/actions/generate-actions.py:35]: Ah, I didn't know this was valid.

Comment (NeoRaider) [reply contrib/actions/generate-actions.py:35]:
```
Artifact name is not valid: ar71xx-tiny/logs. It cannot contain '\', '/', "', ':', '<', '>', '|', '*', and '?'
```
Options:
- `{target_name}.logs`
- `{target_name}@logs`
- `{target_name}+logs`
- `{target_name}_logs`
- `{target_name}~logs`
- ...

Also, it looks like they failed to escape `'` and/or `"` correctly in their error message ;)

Comment (blocktrron) [reply contrib/actions/generate-actions.py:35]: I would sick with `-` then

Comment (NeoRaider) [reply contrib/actions/generate-actions.py:35]: I'm leaning towards `{target_name}_logs` or maybe `{target_name}--logs`. @mweinelt @rotanid ?

Comment (mweinelt) [reply contrib/actions/generate-actions.py:35]: I'm leaning towards anything but `-`. it needs to be visually different from `target-subtarget`.

Comment (blocktrron) [reply contrib/actions/generate-actions.py:35]: Go with `_` then

**Pull request merged**